static_files/scripts/fund.sh

```diff
@@ -20,21 +21,25 @@ send() {
 }
 
 for chain_id in "${chain_ids[@]}"; do
-  proposer_priv=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/1")
-  proposer_addr=$(cast wallet address "$proposer_priv")
-  write_keyfile "$proposer_addr" "$proposer_priv" "proposer-$chain_id"
-  batcher_priv=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/2")
-  batcher_addr=$(cast wallet address "$batcher_priv")
-  write_keyfile "$batcher_addr" "$batcher_priv" "batcher-$chain_id"
-  sequencer_priv=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/3")
-  sequencer_addr=$(cast wallet address "$sequencer_priv")
-  write_keyfile "$sequencer_addr" "$sequencer_priv" "sequencer-$chain_id"
-  challenger_priv=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/4")
-  challenger_addr=$(cast wallet address "$challenger_priv")
-  write_keyfile "$challenger_addr" "$challenger_priv" "challenger-$chain_id"
-  send "$proposer_addr"
-  send "$batcher_addr"
-  send "$challenger_addr"
+  for index in "${!roles[@]}"; do
+    role="${roles[$index]}"
+    role_idx=$((index+1))
+
+    # Skip wallet addrs for anything other Proposer/Batcher/Sequencer/Challenger if not on local L1
+    if [[ "${L1_NETWORK}" != "local" && $role_idx -gt 4 ]]; then
+      continue
+    fi
+
+    private_key=$(cast wallet private-key "$mnemonic" "m/44'/60'/2'/$chain_id/$role_idx")
+    address=$(cast wallet address "${private_key}")
+    write_keyfile "${address}" "${private_key}" "${role}-$chain_id"
+    send "${address}"
+
+    echo "${role} on chain $chain_id, private key:"${private_key}", address:"${address}""
+  done
 
   cat "/network-data/genesis-$chain_id.json" | jq --from-file /fund-script/gen2spec.jq > "/network-data/chainspec-$chain_id.json"
-done
\ No newline at end of file
+done
+
+echo "L1 faucet private key:"0x${PRIVATE_KEY}", address:"${addr}""
```

Comment:
Can you output this as JSON and write it to a file so that it can be parsed by other tools?

Reply:
Done. `wallets.json` now has all the info:

```
 optimism-package git:(contract-addresses) ✗ cat long-crag--23803aa49e3942648aebbdebc75e57c4/files/op-deployer-configs/wallets.json | jq
{
  "proposer_private_key": "0x9d77c0558ed6f31e7be16c2f45fd1f3ec3f66d20526b7fe3961872d38ac98fe7",
  "proposer_address": "0xb0994E702b603df7191cd68E6544f99126135e34",
  "batcher_private_key": "0xb3d2d558e3491a3709b7c451100a0366b5872520c7aa020c17a0e7fa35b6a8df",
  "batcher_address": "0xD3F2c5AFb2D76f5579F326b0cD7DA5F5a4126c35",
  "sequencer_private_key": "0x459f58fd8ef3d9123333514d0f2445153fc29ddc43be239356dad70f506475d9",
  "sequencer_address": "0xbb900Cf56918A2639dAA90c3f7DC5DCD2f5B9935",
  "challenger_private_key": "0x717c53f6d6c266889465d78a885cd0a2e22d41f73e21fa1f07ba5849c82d79c3",
  "challenger_address": "0xf08f610d1956CAAAb34f18e9e0A122E389496529",
  "l2ProxyAdmin_private_key": "0x93be1ab4efe7633d951f19783912817a264f70889cc230e6015627695041744b",
  "l2ProxyAdmin_address": "0x21DD8D15472844b905E603Ddd48A485a630B233b",
  "l1ProxyAdmin_private_key": "0xa8574059a5e36acd4d6ea03e90fe8b5233cc87222b0b396f9b1d6f3e5d6368fa",
  "l1ProxyAdmin_address": "0x39Dc844B8b461c6de6c572cad9585d9f5A8DEa6A",
  "baseFeeVaultRecipient_private_key": "0x227e712ee48868fff4651301e8f366bea05e822db3429470bbc7ed2ee045cb67",
  "baseFeeVaultRecipient_address": "0x9ef74F0e58fdEf70c11E587A45c0a85A3258DCeA",
  "l1FeeVaultRecipient_private_key": "0xe6414f68c9f0d5e3470955a39c1af602a3132f709130fea6304f126a78b7f6aa",
  "l1FeeVaultRecipient_address": "0xA7B30F9e99A248C6D623eaaD7bce81217Bb84038",
  "sequencerFeeVaultRecipient_private_key": "0xc11b0cec865b74505976f6c7fc5736a701f1cdf9e46aa72c02d51c0fe6ec3fed",
  "sequencerFeeVaultRecipient_address": "0x15947bF05662caD8DF55f004c055939b574f18c0",
  "systemConfigOwner_private_key": "0x28978385446dde6ce493da611d6c9ae6193d960c1cb73ecfb25800e9c9017f4c",
  "systemConfigOwner_address": "0x8e44fEbeaE9C2F2Bc89480aE8084328Badc496c2",
  "l1_faucet_private_key": "0xeaba42282ad33c8ef2524f07277c03a776d98ae19f581990ce75becb7cfa1c23",
  "l1_faucet_address": "0x589A698b7b7dA0Bec545177D3963A2741105C7C9",
  "l2_faucet_private_key": "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
  "l2_faucet_address": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
}
```

Reply:
Can you make all of these camelCase? E.g. `l1FaucetAddress`, `l1FeeVaultRecipientAddress`, etc. - weird to mix the underscores.

Reply:
Ah, surely! Fixed now.

```
{
  "proposerPrivateKey": "0x9d77c0558ed6f31e7be16c2f45fd1f3ec3f66d20526b7fe3961872d38ac98fe7",
  "proposerAddress": "0xb0994E702b603df7191cd68E6544f99126135e34",
  "batcherPrivateKey": "0xb3d2d558e3491a3709b7c451100a0366b5872520c7aa020c17a0e7fa35b6a8df",
  "batcherAddress": "0xD3F2c5AFb2D76f5579F326b0cD7DA5F5a4126c35",
  "sequencerPrivateKey": "0x459f58fd8ef3d9123333514d0f2445153fc29ddc43be239356dad70f506475d9",
  "sequencerAddress": "0xbb900Cf56918A2639dAA90c3f7DC5DCD2f5B9935",
  "challengerPrivateKey": "0x717c53f6d6c266889465d78a885cd0a2e22d41f73e21fa1f07ba5849c82d79c3",
  "challengerAddress": "0xf08f610d1956CAAAb34f18e9e0A122E389496529",
  "l2ProxyAdminPrivateKey": "0x93be1ab4efe7633d951f19783912817a264f70889cc230e6015627695041744b",
  "l2ProxyAdminAddress": "0x21DD8D15472844b905E603Ddd48A485a630B233b",
  "l1ProxyAdminPrivateKey": "0xa8574059a5e36acd4d6ea03e90fe8b5233cc87222b0b396f9b1d6f3e5d6368fa",
  "l1ProxyAdminAddress": "0x39Dc844B8b461c6de6c572cad9585d9f5A8DEa6A",
  "baseFeeVaultRecipientPrivateKey": "0x227e712ee48868fff4651301e8f366bea05e822db3429470bbc7ed2ee045cb67",
  "baseFeeVaultRecipientAddress": "0x9ef74F0e58fdEf70c11E587A45c0a85A3258DCeA",
  "l1FeeVaultRecipientPrivateKey": "0xe6414f68c9f0d5e3470955a39c1af602a3132f709130fea6304f126a78b7f6aa",
  "l1FeeVaultRecipientAddress": "0xA7B30F9e99A248C6D623eaaD7bce81217Bb84038",
  "sequencerFeeVaultRecipientPrivateKey": "0xc11b0cec865b74505976f6c7fc5736a701f1cdf9e46aa72c02d51c0fe6ec3fed",
  "sequencerFeeVaultRecipientAddress": "0x15947bF05662caD8DF55f004c055939b574f18c0",
  "systemConfigOwnerPrivateKey": "0x28978385446dde6ce493da611d6c9ae6193d960c1cb73ecfb25800e9c9017f4c",
  "systemConfigOwnerAddress": "0x8e44fEbeaE9C2F2Bc89480aE8084328Badc496c2",
  "l1FaucetPrivateKey": "0xeaba42282ad33c8ef2524f07277c03a776d98ae19f581990ce75becb7cfa1c23",
  "l1FaucetAddress": "0x589A698b7b7dA0Bec545177D3963A2741105C7C9",
  "l2FaucetPrivateKey": "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
  "l2FaucetAddress": "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
}
```